src/Calculus.jl

```diff
@@ -80,15 +80,20 @@ Base.:^(Dx::Differential,y::Number) = unaryOp(Dx -> Dx^y, Dx -> y*Dx^(y-1))(Dx)
 Base.:^(Dx::Differential,y::Int) = unaryOp(Dx -> Dx^y, Dx -> y*Dx^(y-1))(Dx)
 Base.:^(x::Number,Dy::Differential) = unaryOp(Dy -> x^Dy, Dy -> log(x)*x^Dy)(Dy)
 
-Base.exp(Dx::Differential) = unaryOp(exp, exp)(Dx)
-
-Base.log(Dx::Differential) = unaryOp(log, x -> 1/x)(Dx)
-
-Base.sqrt(Dx::Differential) = unaryOp(sqrt, x -> 1/(2*sqrt(x)))(Dx)
-
-Base.sin(Dx::Differential) = unaryOp(sin, x -> cos(x))(Dx)
-
-Base.cos(Dx::Differential) = unaryOp(cos, x -> -sin(x))(Dx)
+# Define differentiation rules for most Base and SpecialFunctions math functions
+for (M, f, arity) in DiffRules.diffrules()
+    if arity == 1 && (M == :Base || M == :SpecialFunctions) && f ∉ [:inv, :+, :-, :abs, :trigamma, :digamma, :invdigamma, :gamma, :lgamma] # [:bessely0, :besselj0, :bessely1, :besselj1]
+        deriv = DiffRules.diffrule(M, f, :x)
+        @eval begin
+            $M.$f(Dx::Differential) = unaryOp($f, x->$deriv)(Dx)
+        end
+    elseif arity == 2 && (M == :Base || M == :SpecialFunctions) && f ∉ [:+, :-, :*, :/]
+        deriv = DiffRules.diffrule(M, f, :x, :y)
+        @eval begin
+            $M.$f(Dx::Differential, Dy::Differential) = binaryOp($f, (x, y)->$deriv)(Dx, Dy)
```

Comment:
I screwed around with this a bit and I can't find a nice clean way to make it work. I think we should just remove this and do by binary functions by hand. There aren't many of them and I think it'll be easier.